pennylane/optimize/rotosolve.py

```diff
@@ -132,340 +212,410 @@ class RotosolveOptimizer:
-    ``optimizer`` and ``optimizer_kwargs`` keyword arguments.
-    As an extra feature, the minimized cost of the intermediate univariate reconstructions can
+    Now we carry out the optimization.
+    The minimized cost of the intermediate univariate reconstructions can
     be read out via ``full_output``, including the cost *after* the full Rotosolve step:
 
-    .. code-block :: python
-
-        param = init_param.copy()
-        cost_rotosolve = []
-        for step in range(num_steps):
-            param, cost, sub_cost = opt.step_and_cost(
-                cost_function,
-                *param,
-                num_freqs=num_freqs,
-                full_output=True,
-            )
-            print(f"Cost before step: {cost}")
-            print(f"Minimization substeps: {np.round(sub_cost, 6)}")
-            cost_rotosolve.extend(sub_cost)
-
-    The optimized values for ``x`` are now stored in ``param`` and (sub)steps-vs-cost can be
-    assessed by plotting ``cost_rotosolve``.
+    >>> param = init_param
+    >>> cost_rotosolve = []
+    >>> for step in range(num_steps):
+    ...     param, cost, sub_cost = opt.step_and_cost(
+    ...         cost_function,
+    ...         *param,
+    ...         nums_frequency=nums_frequency,
+    ...         full_output=True,
+    ...         rot_weights = rot_weights,
+    ...         crot_weights = crot_weights,
+    ...     )
+    ...     print(f"Cost before step: {cost}")
+    ...     print(f"Minimization substeps: {np.round(sub_cost, 6)}")
+    ...     cost_rotosolve.extend(sub_cost)
+    Cost before step: 0.04200821039253547
+    Minimization substeps: [-0.230905 -0.863336 -0.980072 -0.980072 -1.       -1.       -1.      ]
+    Cost before step: -0.9999999990681161
+    Minimization substeps: [-1. -1. -1. -1. -1. -1. -1.]
+    Cost before step: -0.9999999999999996
+    Minimization substeps: [-1. -1. -1. -1. -1. -1. -1.]
+
+    The optimized values for the parameters are now stored in ``param``
+    and the optimization behaviour can be assessed by plotting ``cost_rotosolve``,
+    which include the substeps of the Rotosolve optimization.
     The ``full_output`` feature is available for both, ``step`` and ``step_and_cost``.
 
-    The most general form ``RotosolveOptimizer`` is designed to tackle currently is any
-    trigonometric cost function with integer frequencies up to the given value
-    of ``num_freqs`` per parameter. Not all of the integers up to ``num_freqs`` have to
-    be present in the frequency spectrum. In order to tackle equidistant but non-integer
-    frequencies, we recommend rescaling the argument of the function of interest.
+    In general, the frequencies in a QNode will not be integer-valued, requiring us
+    to provide the ``RotosolveOptimizer`` not only with the number of frequencies
+    but their concrete values. For the example QNode above, this happens if the
+    weights are no longer one:
```

Comment:
I'm curious - if you optimize a non-integer frequency cost function using `num_frequency=`, will it continue to 'work', it just won't converge?

Reply:
so I guess it is up to the user here to identify this themselves?

Reply:
The optimizer will not _check_ the spectrum to be integer-valued. But that's the same with `reconstruct`. It would be like passing the wrong gradient function to gradient descent - the code works but the optimizer probably does not converge